### PR TITLE
chore(ci): add Prettier format check to CI workflow

### LINKED ISSUE #241: Add Prettier Format Check to CI Workflow
## Background:
Currently, our CI workflow does not include a code format check. Adding this step would help maintain code consistency and reduce unnecessary discussions during code reviews.

## Proposed Solution:
I propose to add a prettier format check to our CI workflow. This would ensure that all pull requests are automatically validated for code style before being merged.

### PR DESCRIPTION
**Pull request by dayongkr**
This pull request adds a prettier format check to our CI workflow, ensuring that only modified {js, ts, tsx} files are checked against the formatting rules. The comparison is made with the origin/main branch.

close #241 

> Additionally, there are currently many unformatted files, so we will also need a pull request to format these files using the yarn format command.

### CODE REVIEW

**Comment by dayongkr** [comment]
And I also make lint and formatcheck job  not to detact tsx files

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,7 +31,16 @@ jobs:
...
+          command: yarn eslint -c .eslintrc.js $(git diff --name-only --diff-filter=ACMRUXB origin/main | grep -E "(.js$|.ts$)")
+  formatcheck:
+    docker:
+      - image: cimg/node:20.12
+    steps:
+      - checkout
+      - setup
+      - run:
+          name: Format Check
+          command: yarn prettier -c $(git diff --name-only --diff-filter=ACMRUXB origin/main | grep -E "(.js$|.ts$)")
```
Comment (raon0211) [inline .circleci/config.yml:43]:
```suggestion
          command: yarn eslint -c .eslintrc.js $(git diff --name-only --diff-filter=ACMRUXB origin/main | grep -E "(.js$|.ts$)")
      - run:
          name: Prettier
          command: yarn prettier -c $(git diff --name-only --diff-filter=ACMRUXB origin/main | grep -E "(.js$|.ts$)")
```

What about we merge the `lint` and `format` like this? 

I guess it does not have to be run in separate containers.

Comment (dayongkr) [reply .circleci/config.yml:43]: I agree that!

**Comment by dayongkr** [comment]
It has a issue when we dont have changed file then prettier throw a error. So I am finding a solution now.

**Comment by dayongkr** [comment]
> It has a issue when we dont have changed file then prettier throw a error. So I am finding a solution now.

Done

**Comment by raon0211** [comment]
image

Prettier seems to fail. Could you check why?

**Comment by dayongkr** [comment]
> image
> Prettier seems to fail. Could you check why?

I change the logic(using `xargc -r`) for simplify but that command makes fails because `xargc -r` return exit with 1 status code when modified files doesn't exist.

so I rollback that logic.

**Review by raon0211 — APPROVED** [review]

**Pull request merged**